sklearn/cluster/tests/test_k_means.py

```diff
@@ -94,21 +91,21 @@ def test_relocated_clusters(array_constr, algo):
     kmeans.fit(X)
 
     assert_array_equal(kmeans.labels_, expected_labels)
-    assert_almost_equal(kmeans.inertia_, expected_inertia)
-    assert_array_almost_equal(kmeans.cluster_centers_, expected_centers)
+    assert_allclose(kmeans.inertia_, expected_inertia)
+    assert_allclose(kmeans.cluster_centers_, expected_centers)
     assert kmeans.n_iter_ == expected_n_iter
 
 
-@pytest.mark.parametrize("representation", ["dense", "sparse"])
-def test_relocate_empty_clusters(representation):
+@pytest.mark.parametrize("array_constr", [np.array, sp.csr_matrix],
+                         ids=["dense", "sparse"])
+def test_relocate_empty_clusters(array_constr):
     # test for the _relocate_empty_clusters_(dense/sparse) helpers
 
     # Synthetic dataset with 3 obvious clusters of different sizes
     X = np.array(
         [-10., -9.5, -9, -8.5, -8, -1, 1, 9, 9.5, 10]).reshape(-1, 1)
-    if representation == "sparse":
-        X = sp.csr_matrix(X)
-    sample_weight = np.full(shape=10, fill_value=1.)
+    X = array_constr(X)
```

Comment:
For all those ones, we could use the `_convert_container` helper function from the `_testing.py` module.
I think that we should add a docstring to it

Reply:
I'd rather not use it. For these tests we'd only need it for a small subset of its possibilities, and in the same time it misses the possibility to specify dtype for array construction.